File: C++/linearsearch.cpp
/*
Program to search an element in an array using Linear Search
Author: Anuj Singh
Date Modified: 4th October 2021
*/


#include<iostream>
using namespace std;
int main()
{
  int n;
  cin>>n;
  int a[n];
  int ans; 
  for(int i=0;i<n;i++){
    cin>>a[i];
  }
  int fn;
  cout<<"enter a number to search"<<endl;
  cin>>fn;
  for(int i=0;i<n;i++){
    if(a[i]==fn){
      ans=i;
      break;
    }
  }
    cout<<ans<<endl;
    return 0;
  }
      

File: JAVA/Linear_Search.java
/*
  Code: Linear Search (JAVA)
  Name: Shreeyanshi Gautam
  Date of Modification: 31/10/2021
 */
import java.util.Scanner;
class LinearSearch
{
   public static void main(String args[])
   {
      int i, n, ele, array[];
      Scanner sc = new Scanner(System.in);
      System.out.print("Enter the number of elements: ");
      n= sc.nextInt(); 
      array = new int[n]; 
      System.out.println("Enter "+ n +" elements: ");
      for (i= 0;i < n;i++) 
      {
        array[i] = sc.nextInt();
      }
      System.out.print("Enter the element to be searched: ");
      ele= sc.nextInt();
      for (i = 0;i < n;i++)
      {
         if (array[i] == ele) 
         {
           System.out.println("Element found at position: "+(i+1));
           break;
         }
      }
      if (i == n)
        System.out.println("Element was not found in the array");
   }
}
